src/vmm/src/device_manager/persist.rs

```diff
@@ -611,14 +646,74 @@ mod tests {
         let vmm = default_vmm();
         let device_states: DeviceStates =
             DeviceStates::deserialize(&mut buf.as_slice(), &version_map, 2).unwrap();
+        let vm_resources = &mut Default::default();
         let restore_args = MMIODevManagerConstructorArgs {
             mem: vmm.guest_memory().clone(),
             vm: vmm.vm.fd(),
             event_manager: &mut event_manager,
+            for_each_restored_device: VmResources::update_from_restored_device,
+            vm_resources,
         };
         let restored_dev_manager =
             MMIODeviceManager::restore(restore_args, &device_states).unwrap();
 
+        let expected_vm_resources = format!(
+            r#"{{
+  "balloon": {{
```

Comment:
the indentation is a bit off

Reply:
If I change the indentation here, the test will fail since it's a raw string and the indentation of the raw string JSON will no longer match the pretty printed JSON from serde serialization. Is there another way to change the indentation formatting but still have the assertion pass?